qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -0,0 +1,348 @@
+            \hat{\mathcal{A}}\vert \beta \rangle`. For a two-particle operator this
+            array will have shape ``(me_table.shape[0], 5)`` with each row containing
+            the indices :math:`\alpha`, :math:`\beta`, :math:`\gamma`, :math:`\delta` and
+            the matrix elements :math:`\langle \alpha, \beta \vert \hat{\mathcal{B}}
+            \vert \gamma, \delta \rangle`.
+        init_term: the contribution of doubly-occupied orbitals, if any, or other quantity
+            required to initialize the many-body observable.
+        mapping (str): specifies the fermion-to-qubit mapping. Input values can
+            be ``'jordan_wigner'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+        pennylane.Hamiltonian: the fermionic-to-qubit transformed observable
+
+    **Example**
+
+    >>> s2_matrix_elements, init_term = get_spin2_matrix_elements('h2', './pyscf/sto-3g')
+    >>> s2_obs = observable(s2_matrix_elements, init_term=init_term)
+    >>> print(s2_obs)
+    (0.75) [I0]
+    + (0.375) [Z1]
+    + (-0.375) [Z0 Z1]
+    + (0.125) [Z0 Z2]
+    + (0.375) [Z0]
+    + (-0.125) [Z0 Z3]
+    + (-0.125) [Z1 Z2]
+    + (0.125) [Z1 Z3]
+    + (0.375) [Z2]
+    + (0.375) [Z3]
+    + (-0.375) [Z2 Z3]
+    + (0.125) [Y0 X1 Y2 X3]
+    + (0.125) [Y0 Y1 X2 X3]
+    + (0.125) [Y0 Y1 Y2 Y3]
+    + (-0.125) [Y0 X1 X2 Y3]
+    + (-0.125) [X0 Y1 Y2 X3]
+    + (0.125) [X0 X1 X2 X3]
+    + (0.125) [X0 X1 Y2 Y3]
+    + (0.125) [X0 Y1 X2 Y3]
+    """
+
+    if mapping.strip().lower() not in ("jordan_wigner", "bravyi_kitaev"):
+        raise TypeError(
+            "The '{}' transformation is not available. \n "
+            "Please set 'mapping' to 'jordan_wigner' or 'bravyi_kitaev'.".format(mapping)
+        )
+
+    sp_op_shape = (3,)
+    tp_op_shape = (5,)
+    for i_table in me_table:
+        if np.array(i_table).shape not in (sp_op_shape, tp_op_shape):
+            raise ValueError(
+                "expected entries of 'me_table' to be of shape (3,) or (5,) ; got {}".format(
+                    np.array(i_table).shape
+                )
+            )
+
+    # Initialize the FermionOperator
+    mb_obs = FermionOperator() + FermionOperator("") * init_term
+
+    for i in me_table:
+
+        if i.shape == (5,):
+            # two-particle operator
+            mb_obs += FermionOperator(
+                ((int(i[0]), 1), (int(i[1]), 1), (int(i[2]), 0), (int(i[3]), 0)), i[4]
+            )
+        elif i.shape == (3,):
+            # single-particle operator
+            mb_obs += FermionOperator(((int(i[0]), 1), (int(i[1]), 0)), i[2])
+
+    # Map the fermionic operator to a qubit operator
+    if mapping.strip().lower() == "bravyi_kitaev":
+        return structure.convert_observable(bravyi_kitaev(mb_obs))
+
+    return structure.convert_observable(jordan_wigner(mb_obs))
+
+
+def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
```

Comment:
I might have missed this previously, but what is the difference between `spin2_matrix_elements` and `get_spin2_matrix_elements`? The `get` doesn't seem to differentiate them much for me, since I expect that `spin2_matrix_elements` also 'gets' the matrix element.

Reply:
The difference is that ``spin2_matrix_elements`` implement the expression(s) to compute the matrix elements and ``get_spin2_matrix_elements`` loads the HF electronic structure, defines an active space and then calls ``spin2_matrix_elements`` to get the matrix elements computed in this sub-space.
For simpler observables like 'S_z' or 'N' we don't need a separate function to compute the matrix elements as they are very simple but for other obserbables we will need independent functions.

The important functions from a user perspective are the ones with the ``get_`` prefix.
Should we make functions like ``spin2_matrix_elements`` private -> ``_spin2_matrix_elements`` ?